protected/controllers/AdminFileController.php

```diff
@@ -464,41 +464,51 @@ public function actionEditAttr()
         echo CJSON::encode(array('success' => false));
     }
 
+    /**
+     * Yii's method for routing urls to an action. Override to use custom actions
+     */
+    public function actions()
+    {
+        return array(
+            'deleteFileAttribute'=>'application.controllers.adminFile.DeleteFileAttributeAction',
+        );
+    }
+
     /**
      * Deletes a particular model.
      * If deletion is successful, the browser will be redirected to the 'admin' page.
      * @param integer $id the ID of the model to be deleted
      */
    public function actionDelete($id)
     {
-        
+
         if (Yii::app()->request->isPostRequest) {
             // we only allow deletion via POST request
-           $file = File::model()->findByPk($id);
-           
-          // $file->fileSamples->delete();
+            $file = File::model()->findByPk($id);
+            // $file->fileSamples->delete();
           foreach ($file->fileAttributes as $fileattributes) {
               print_r($fileattributes);
               $fileattributes->delete();
-              
+
           }
          foreach ($file->fileSamples as $filesample) {
               print_r($filesample);
               $filesample->delete();
-              
+
           }
-      
-           $file->delete();
-           
-            
+
+         $file->delete();
+
+        }
 
             // if AJAX request (triggered by deletion via admin grid view), we should not redirect the browser
-           
-            if (!isset($_GET['ajax']))
-                $this->redirect(isset($_POST['returnUrl']) ? $_POST['returnUrl'] : array('admin'));
-        } else
+
+        if (!isset($_GET['ajax'])) {
+            $this->redirect(isset($_POST['returnUrl']) ? $_POST['returnUrl'] : array('admin'));
+        } else {
             throw new CHttpException(400, 'Invalid request. Please do not repeat this request again.');
-      
+        }
+
     }
```

Comment:
Are you sure the new added curly braces are required here?

Reply:
I think they are required, otherwise there will have syntax error.